### DISCUSSION TITLE
VMAF for VVC

### DISCUSSION THREAD

**Discussion by andini2807** [fraunhoferhhi/vvenc#89] [Q&A]
Hi, my name is Antonio and I am doing a final Master thesis. I need to get the VMAF value of an VVC video stream. How I can do it? Thank you.

**Comment by alexander-g2** [comment]
VVenC doesn't report VMAF scores. Use [VVdeC](https://github.com/fraunhoferhhi/vvdec) for decoding, and [VMAF](https://github.com/Netflix/vmaf) for measuring quality.

Encoding:
`vvencapp -s 640x360 -r 60 -c yuv420_10 -o VVC.266 -i source.yuv`

Decoding:
`vvdecapp -b VVC.266 -o VVC.yuv`

Measuring quality:
`vmaf -r source.yuv -d VVC.yuv -w 640 -h 360 -p 420 -b 10`

**Reply by andini2807** [reply]
https://user-images.githubusercontent.com/47917100/130665830-070b38b2-6647-44e8-aa94-a183f6e8c51d.mp4

Hello, I have an issue trying to get the VMAF score of an 1920x1080 10 seg, 30 fps video.
I used this command for encoding:
`vvencapp -i Football.yuv -s 1920x1080 -c yuv420 -r 30 --preset medium --qp 31 --qpa 0 -ip 64 -t 4 -o Football_vvc_qp31.266`
This one for decoding:
`vvdecapp -b Football_vvc_qp31.266 -o Football_vvc.yuv`
and this one to get the VMAF:
`vmaf -r Football.yuv -d prueba.yuv -w 1920 -h 1080 -p 420 -b 8`
and I get a VMAF of 6 points. Just to check I encoded the Football_vvc.yuv file with h264 and the video had 20 seconds duration and horrible quality. What I am doing wrong? Thank you.

**Reply by alexander-g2** [reply]
Add `--internal-bitdepth 8` to your VVenC command. 

`vvencapp -i Football.yuv -s 1920x1080 -c yuv420 --internal-bitdepth 8 -r 30 --preset medium --qp 31 --qpa 0 -ip 64 -t 4 -o Football_vvc_qp31.266`

Use `vvencapp --fullhelp` for more info.

**Reply by andini2807** [reply]
I changed my video to bitdepth 10 and now works perfectly. I will try --internal-bitdepth 8 too. Thank you very much. You help me a lot.
